build/util/src/lib.rs

```diff
@@ -56,7 +56,7 @@ pub fn has_feature(s: &str) -> bool {
 ///
 /// This will set one of `cfg(armv6m)`, `cfg(armv7m)`, or `cfg(armv8m)`
 /// depending on the value of the `TARGET` environment variable.
-pub fn expose_m_profile() {
+pub fn expose_m_profile() -> Result<()> {
```

Comment:
This change was necessary because I added an `examples/` dir to `ringbuf`, and the `cargo test` run on CI for the host target attempts to build examples as well as running tests. Since `ringbuf` depends on `static_cell` (and now, on `armv6m-atomic-hack` as well), building `ringbuf` calls `expose_m_profile`, which panics on x86_64 build hosts. Therefore, I've changed this method to return a `Result` instead, which the libs that _can_ reasonably be built for x86_64 (`ringbuf`, `static-cell`, `armv6m-atomic-hack`) turn into a warning, as we only need to determine ARM profile for ARM targets, while the crates that only build for ARM unwrap, maintaining the previous behavior of failing the build if built for a non-ARM target.

This, unfortunately, required changing a bunch of build scritps as well. Sorry about the big diff!